p4-16/spec/P4-16-draft-spec.mdk

```diff
@@ -5444,41 +5458,30 @@ packet_out {
         this.data.clear();
         this.lengthInBits = 0;
     }
-    // append entire header if it is valid
-    // T must be a header type
-    void emit<T>(T header) {
-        this.emit(header.valid$, header);
-    }
-
-    // append the data to the packet if the condition is true
-    void emit<T>(bool cond, T data) {
-        if (!cond) return;
-        this.data.append(data);
+    // append the data to the packet
+    void raw_emit<T>(T data) {
+        really this.data.append(data);
         this.lengthInBits += data.lengthInBits;
     }
-}
+    // append data to the packet if valid
+    void emit<T>(T data) {
```

Comment:
Added "We also use standard `for` notation to
iterate through the elements of a stack `(e : data)` and list of
fields for header stacsk and unions `(f : data.fields$)`."